frontend/src/app/services/university-search.service.ts

```diff
@@ -0,0 +1,99 @@
+import { Injectable } from '@angular/core';
+import { Http } from '@angular/http';
+import 'rxjs/add/operator/map';
+import 'rxjs/add/operator/toPromise';
+import 'rxjs/add/operator/share';
+import { University } from '../models/university';
+import {environment} from '../../environments/environment';
+import {SubjectHubApiResponse} from "../models/subject-hub-api-response";
+
+@Injectable()
+export class UniversitySearchService {
+    constructor(private http: Http) {}
+
+    fetch(): Promise<SubjectHubApiResponse<University[]>> {
+        return this.universitiesAction(`/universities`,
+            `No universities found`, `Universities request failed`);
+    }
+
+    searchByNameAndAbbreviation(name: string, abbreviation: string): Promise<SubjectHubApiResponse<University[]>> {
```

Comment:
We don't need 4 different methods for search.

Should simply be getUniversities(name: string, abbreviation: string) and you should add them to the http request using [HttpParams](https://angular.io/api/common/http/HttpParams) if they aren't null.